client/fingerprint/cpu.go

```diff
@@ -51,101 +45,154 @@ func (f *CPUFingerprint) Fingerprint(request *FingerprintRequest, response *Fing
-	if modelName := stats.CPUModelName(); modelName != "" {
-		response.AddAttribute("cpu.modelname", modelName)
-		f.logger.Debug("detected CPU model", "name", modelName)
+	if model := cpuid.CPU.BrandName; model != "" {
+		response.AddAttribute("cpu.modelname", model)
+		f.logger.Debug("detected CPU model", "name", model)
 	}
 }
 
-func (*CPUFingerprint) frequency(mhz uint64) string {
-	return fmt.Sprintf("%.0f", float64(mhz))
+func (*CPUFingerprint) frequency(mhz numalib.MHz) string {
+	return strconv.FormatUint(uint64(mhz), 10)
 }
 
 func (f *CPUFingerprint) setFrequency(response *FingerprintResponse) {
-	power, efficiency := stats.CPUMHzPerCore()
+	performance, efficiency := f.top.CoreSpeeds()
 	switch {
 	case efficiency > 0:
 		response.AddAttribute("cpu.frequency.efficiency", f.frequency(efficiency))
-		response.AddAttribute("cpu.frequency.power", f.frequency(power))
+		response.AddAttribute("cpu.frequency.performance", f.frequency(performance))
 		f.logger.Debug("detected CPU efficiency core speed", "mhz", efficiency)
-		f.logger.Debug("detected CPU power core speed", "mhz", power)
-	case power > 0:
-		response.AddAttribute("cpu.frequency", f.frequency(power))
-		f.logger.Debug("detected CPU frequency", "mhz", power)
+		f.logger.Debug("detected CPU performance core speed", "mhz", performance)
+	case performance > 0:
+		response.AddAttribute("cpu.frequency", f.frequency(performance))
+		f.logger.Debug("detected CPU frequency", "mhz", performance)
 	}
 }
 
 func (*CPUFingerprint) cores(count int) string {
 	return strconv.Itoa(count)
 }
 
+func (*CPUFingerprint) nodes(count int) string {
+	return strconv.Itoa(count)
+}
+
 func (f *CPUFingerprint) setCoreCount(response *FingerprintResponse) {
-	power, efficiency := stats.CPUNumCores()
+	total := f.top.NumCores()
+	performance := f.top.NumPCores()
+	efficiency := f.top.NumECores()
 	switch {
 	case efficiency > 0:
 		response.AddAttribute("cpu.numcores.efficiency", f.cores(efficiency))
-		response.AddAttribute("cpu.numcores.power", f.cores(power))
+		response.AddAttribute("cpu.numcores.performance", f.cores(performance))
+		response.AddAttribute("cpu.numcores", f.cores(total))
 		f.logger.Debug("detected CPU efficiency core count", "cores", efficiency)
-		f.logger.Debug("detected CPU power core count", "cores", power)
-	case power > 0:
-		response.AddAttribute("cpu.numcores", f.cores(power))
-		f.logger.Debug("detected CPU core count", power)
+		f.logger.Debug("detected CPU performance core count", "cores", performance)
+		f.logger.Debug("detected CPU core count", "cores", total)
+	default:
+		response.AddAttribute("cpu.numcores", f.cores(total))
+		f.logger.Debug("detected CPU core count", "cores", total)
 	}
-	f.nodeResources.Cpu.TotalCpuCores = uint16(power + efficiency)
+	f.nodeResources.Cpu.TotalCpuCores = uint16(total)
 }
 
-func (f *CPUFingerprint) setReservableCores(request *FingerprintRequest, response *FingerprintResponse) {
-	reservable := request.Config.ReservableCores
```

Comment:
It looks like we've dropped handling of user-provided configuration here, but we specifically refer to it in the new numa package. Should this still be part of the fingerprint?

Reply:
Filed https://github.com/hashicorp/nomad/issues/18213 - will think about this more